packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -386,6 +317,106 @@ library LibProving {
         });
     }
 
+    // TODO(brecht): can we write a circuit for this function?
```

Comment:
Yeah sure, doing it in the smart contract just easier and I thought would have been cheap enough to not make much of a difference, but it's more expensive than I expected so seems to be worth it do it in the circuit (well unless maybe the cheap precompile gets added to Ethereum to do this).